source/Plugins.cpp

```diff
@@ -59,42 +59,48 @@ bool Plugin::IsValid() const
 
 
 
-// Try to load a plugin at the given path. Returns true if loading succeeded.
+// Attempt to load a plugin at the given path.
 const Plugin *Plugins::Load(const string &path)
 {
 	// Get the name of the folder containing the plugin.
 	size_t pos = path.rfind('/', path.length() - 2) + 1;
 	string name = path.substr(pos, path.length() - 1 - pos);
 
-	auto *plugin = plugins.Get(name);
-
 	string pluginFile = path + "plugin.txt";
+	string aboutText;
 
 	// Load plugin metadata from plugin.txt.
-	DataFile file(pluginFile);
-	for(const DataNode &child : file)
+	bool hasName = false;
+	for(const DataNode &child : DataFile(pluginFile))
 	{
 		if(child.Token(0) == "name" && child.Size() >= 2)
-			plugin->name = child.Token(1);
+		{
+			name = child.Token(1);
+			hasName = true;
+		}
 		else if(child.Token(0) == "about" && child.Size() >= 2)
-			plugin->aboutText = child.Token(1);
+			aboutText = child.Token(1);
+		else
+			child.PrintTrace("Skipping unrecognized attribute:");
 	}
 
-	// Set missing required values.
-	if(plugin->name.empty())
+	// 'name' is a required field for plugins with a plugin description file.
+	if(Files::Exists(pluginFile) && !hasName)
+		Logger::LogError("Warning: Missing required \"name\" field inside plugin.txt");
+
+	// Plugin names should be unique.
+	auto *plugin = plugins.Get(name);
```

Comment:
What about the case where a plugin doesn't have a `plugins.txt` file?
Shouldn't you be assigning the folder name to `name`? It seems like you're searching `plugins` with an empty string otherwise.

Reply:
The default value of `name` is the folder name yes